release/models/bgp/openconfig-bgp-policy.yang

```diff
@@ -163,6 +170,29 @@ module openconfig-bgp-policy {
 
   // typedef statements
 
+  typedef bgp-set-as-path-option-type {
+    type enumeration {
+      // TODO(xqm): maybe use prepend instead?
+      enum ADD {
```

Comment:
I am not so sure about certain behaviors after thinking about this a little more. Unlike communities, as-path is ordered and therefore could be a bit tricky.

1. I wonder what would the expected behavior for `add as-path "1 2 3"`. Do we prepend "1 2 3" to the AS path? How about `delete as-path "1 2 3"? Do we delete 1) every AS1, AS2 and AS3 from the path 2) all "1 2 3" sub-paths from the as-path?
2. Using as-path-set reference as input might complicate things further. For instance, an as-path-set can have multiple members such as `["1 2", "3 4 5"]`. After `replace as-path-set`, will the AS path become `"1 2 3 4 5"`? If there are regex in as-path-set, maybe it works for delete, but it's more like invalid input for add/replace.

Would like to have your thoughts to better define the semantics in these cases. Thanks Darren!

Reply:
Regarding precedence, I am aware of as path prepending 'n' times, adding the "last-asn" to the list and replacing the originating asn with one's own asn (as-override) .   

I'm not aware of existing implementations that allow arbitrary add/remove operations from an asn path.  

Regarding "editing" an as-path set, practically in  gNMI (although OC isn't gnmi only, netconf is also used),  there is no 'list editing / insert' operation.  To change an 'ordered-by-user' list in gnmi, the whole list must be replaced.